### ISSUE TITLE
Install-SqlWhoIsActive - Database 'Downloading sp_WhoIsActive zip file, unzipping and installing. master' does not exist

### ISSUE THREAD

**Issue by dwierenga** [sqlcollaborative/dbatools#216]
When running a SQL 2012 migration from one clustered instance to another, I got this message in the `dbatools-exceptions.txt` logfile.  It does not seem fatal, but it does seem like a very odd message.

```
9/21/2016 12:05:21 PM
System.Management.Automation.MethodInvocationException: Exception calling "ChangeDatabase" with "1" argument(s): "Database 'Downloading sp_WhoIsActive zip file, unzipping and installing. master' does not exist. Make sure that the name is entered correctly." ---> System.Data.SqlClient.SqlException: Database 'Downloading sp_WhoIsActive zip file, unzipping and installing. master' does not exist. Make sure that the name is entered correctly.
   at System.Data.SqlClient.SqlConnection.OnError(SqlException exception, Boolean breakConnection, Action`1 wrapCloseInAction)
   at System.Data.SqlClient.TdsParser.ThrowExceptionAndWarning(TdsParserStateObject stateObj, Boolean callerHasConnectionLock, Boolean asyncClose)
   at System.Data.SqlClient.TdsParser.TryRun(RunBehavior runBehavior, SqlCommand cmdHandler, SqlDataReader dataStream, BulkCopySimpleResultSet bulkCopyHandler, TdsParserStateObject stateObj, Boolean& dataReady)
   at System.Data.SqlClient.TdsParser.Run(RunBehavior runBehavior, SqlCommand cmdHandler, SqlDataReader dataStream, BulkCopySimpleResultSet bulkCopyHandler, TdsParserStateObject stateObj)
   at System.Data.SqlClient.SqlInternalConnectionTds.ChangeDatabaseInternal(String database)
   at System.Data.SqlClient.SqlInternalConnection.ChangeDatabase(String database)
   at System.Data.SqlClient.SqlConnection.ChangeDatabase(String database)
   at CallSite.Target(Closure , CallSite , Object , Object )
   --- End of inner exception stack trace ---
   at System.Management.Automation.ExceptionHandlingOps.CheckActionPreference(FunctionContext funcContext, Exception exception)
   at System.Management.Automation.Interpreter.ActionCallInstruction`2.Run(InterpretedFrame frame)
   at System.Management.Automation.Interpreter.EnterTryCatchFinallyInstruction.Run(InterpretedFrame frame)
   at System.Management.Automation.Interpreter.EnterTryCatchFinallyInstruction.Run(InterpretedFrame frame)
9/21/2016 3:46:01 PM
System.Management.Automation.MethodInvocationException: Exception calling "ExecuteNonQuery" with "1" argument(s): "An exception occurred while executing a Transact-SQL statement or batch." ---> Microsoft.SqlServer.Management.Common.ExecutionFailureException: An exception occurred while executing a Transact-SQL statement or batch. ---> System.Data.SqlClient.SqlException: RegCreateKeyEx() returned error 5, 'Access is denied.'
   at Microsoft.SqlServer.Management.Common.ConnectionManager.ExecuteTSql(ExecuteTSqlAction action, Object execObject, DataSet fillDataSet, Boolean catchException)
   at Microsoft.SqlServer.Management.Common.ServerConnection.ExecuteNonQuery(String sqlCommand, ExecutionTypes executionType)
   --- End of inner exception stack trace ---
   at Microsoft.SqlServer.Management.Common.ServerConnection.ExecuteNonQuery(String sqlCommand, ExecutionTypes executionType)
   at CallSite.Target(Closure , CallSite , Object , Object )
   --- End of inner exception stack trace ---
   at System.Management.Automation.ExceptionHandlingOps.CheckActionPreference(FunctionContext funcContext, Exception exception)
   at System.Management.Automation.Interpreter.ActionCallInstruction`2.Run(InterpretedFrame frame)
   at System.Management.Automation.Interpreter.EnterTryCatchFinallyInstruction.Run(InterpretedFrame frame)
   at System.Management.Automation.Interpreter.EnterTryCatchFinallyInstruction.Run(InterpretedFrame frame)

```

**Comment by ctrlbold** [comment]
hey @dwierenga, thanks for logging this. The first issue was run  at 9/21/2016 12:05:21 PM and the second one was at 3:46 which is when I'm assuming you ran the Start-SqlMigration. This is an issue that needs to be fixed in Oct. I get it as well. It's a change to the SQL Agent that sometimes works, and sometimes doesn't. 

I also need to update Write-Exception with more info, but I may have already done that and it may be in this month's batch.

**Comment by ctrlbold** [comment]
I haven't had caffeine this morning yet ;) So to clarify, the sp_whoisactive issue did not come from the migration command, which was run hours later.  That is a strange issue for the sp_whoisactive, however. I'll take a closer look at the sp_whoisactive files and see what's up. October is fix bugs month for me.

- [ ] Investigate recent addition to SQL Agent migration that is causing an access denied sometimes
- [ ] Check to ensure that Write-Exception has the additional info needed 
- [ ] Check to see why sp_whoisactive gives that dumb output

**Comment by dwierenga** [comment]
@ctrlbold ahh I see, I didn't realize that all of the cmdlets logged to the exception file, not just Start-SQLMigration.  I definitely ran Install-SqlWhoIsActive several hours prior to the migration and got an error on it; I just hadn't realized it had logged that error to the exception file.

**Comment by ctrlbold** [comment]
This has been addressed in #218!
